### ISSUE TITLE
yourtv.pl - Plex handling of tag episode-num system="original-air-date" causes incorrect episode numbering

### ISSUE THREAD

**Issue by eavent-splunk** [markcs/xml_tv#19]
So I've found that using yourtv.pl to generate XML data for Plex results in incorrect detection of episode numbers, and possibly misclassification of movies as TV shows.

Here's an image of tonight's guide in plex - you can see some shows have episode numbers like "E1011" (today's date), and those after midnight switch to "E1012". Others have no issue:
![image](https://user-images.githubusercontent.com/39285196/66652127-a0d82680-ec80-11e9-8a33-83371fdc1997.png)

A quick spot check in the XML output shows affected shows have metadata like this:
```
<programme start="20191012020500 +1100" stop="20191012023300 +1100" channel="22.yourtv.com.au">
<title>Red Dwarf</title>
<sub-title>Demons And Angels</sub-title>
<desc>An out-of-control triplicator creates two more Red Dwarfs. They are complete with replica crews - one divine, the other evil as can be.</desc>
<category>Comedy</category>
<icon src="https://s3-ap-southeast-2.amazonaws.com/nine-tvmg-images-prod/27/35/20/273520_red_dwarf.jpg" />
<episode-num system="SxxExx">S2019E1012</episode-num>
<episode-num system="xmltv_ns">2018.1011.</episode-num>
<episode-num system="imdb.com">series/0094535</episode-num>
<episode-num system="original-air-date">2019-10-12 02:05:00</episode-num>
<previously-shown start="2019-10-12" />
<rating><value>PG</value></rating>
</programme>
```

While unaffected have this:
```
<programme start="20191011235000 +1100" stop="20191012001500 +1100" channel="31.yourtv.com.au">
<title>The Last Man on Earth</title>
<sub-title>Skidmark</sub-title>
<desc>Tandy tries to host a party for the group, and Todd's luck takes a turn for the better.</desc>
<category>Comedy</category>
<icon src="https://s3-ap-southeast-2.amazonaws.com/nine-tvmg-images-prod/42/90/22/429022_591632_SERIES_55917_i_T2_t1_.jpg" />
<episode-num system="SxxExx">S02E14</episode-num>
<episode-num system="xmltv_ns">1.13.</episode-num>
<episode-num system="imdb.com">series/3230454</episode-num>
<rating><value>M</value></rating>
</programme>
```

Similarly the same <episode-num system="original-air-date"> tag in movies seems to result in Plex considering them to be TV shows (as it gives them an episode number), not movies. This means they don't display under "Upcoming Movies" in the "Live TV & DVR" view, and default to the TV Show library. Example:
Displays as TV Show: 
```
<programme start="20191010210000 +1100" stop="20191011000000 +1100" channel="7.yourtv.com.au">
<title>Kingsman: The Golden Circle</title>
<desc>When the Kingsman headquarters are destroyed and the world is held hostage, their journey leads them to the discovery of an allied spy organisation in the US called Statesman.</desc>
<category>Action</category>
<category>Cinema movie</category>
<episode-num system="imdb.com">series/4649466</episode-num>
<episode-num system="original-air-date">2019-10-10 21:00:00</episode-num>
<previously-shown start="2019-10-10" />
<rating><value>MA</value></rating>
</programme>
```
Displays as Movie:
```
<programme start="20191012190000 +1100" stop="20191012220000 +1100" channel="9.yourtv.com.au">
<title>Star Wars: Episode II - Attack of the Clones</title>
<desc>Ten years after initially meeting, Anakin Skywalker shares a forbidden romance with Padme Amidala, while Obi-Wan Kenobi investigates an assassination attempt on the senator and discovers a secret clone army.</desc>
<category>Sci-Fi</category>
<category>Cinema movie</category>
<icon src="http://img-store-prod.switch.tv/images/d1c81e1d-586e-4796-bef0-8e98554bc2cc/611157_91.png" />
<episode-num system="imdb.com">series/0121765</episode-num>
<rating><value>PG</value></rating>
</programme>
```

It's possible this might be considered a Plex or YourTV issue, but I thought it worth mentioning if it could be correct in code. Cheers!

**Comment by markcs** [comment]
A number of different issues here and it is probably due to the data provided by yourtv.

Plex considers everything as a movie by default.  If the xml includes the episode-num, then Plex will identify this as a TV show/series and not a movie.  The script identifies this by the 'programType' tag in the json.  

Are you still seeing this issue?

**Comment by eavent-splunk** [comment]
Just updated plex to 1.18.1.1973 and did a quick spot-check. Red Dwarf tonight has a proper tag:
`<episode-num system="SxxExx">S07E06</episode-num>`

But right after it is Peep show, which should be S06E03 but has:
`<episode-num system="SxxExx">S2019E1029</episode-num>`

So this time I checked on yourtv, and lo and behold:
https://www.yourtv.com.au/program/peep-show/125101/
![image](https://user-images.githubusercontent.com/39285196/67763988-00b23800-fa9d-11e9-89f7-f5d4631ee080.png)

The showing for tonight doesn't display any episode data. So that clears that up, it is just a yourTV data issue.

Regarding movies as TV shows, I think this is happening for sequels. For example, ">The Second Best Exotic Marigold Hotel" has this in XML:
`<episode-num system="imdb.com">series/2555736</episode-num>
<episode-num system="original-air-date">2019-10-29 20:30:00</episode-num>`

The second line I'm not sure the point of, but the first seems reasonable given this is a sequel. I guess Plex just expects different tagging for movie series, not much to be done about that unless you feel like re-writing the XML as it comes in to make Plex happy (I wouldn't). Closing out the issue.
